pkg/monitortestlibrary/podaccess/pod_log_streamer.go

```diff
@@ -133,10 +147,13 @@ func (s *PodStreamer) streamLogsReader(ctx context.Context, currPod *corev1.Pod)
 		Timestamps: true,
 	}).Stream(context.Background())
 	if err != nil {
-		s.errs <- LogError{
-			Pod:     currPod,
-			Locator: locator,
-			Error:   err,
+		// don't attempt to write the error if we have been stopped
+		if ctx.Err() == nil {
```

Comment:
if the CPU scheduler parked this thread after the check prior to the channel write, this still races, correct?